adminSiteClient/DataInsightIndexPage.tsx

```diff
@@ -428,6 +456,13 @@ export function DataInsightIndexPage() {
         )
     }, [admin])
 
+    useEffect(() => {
+        const fetchTags = async () =>
+            (await admin.getJSON("/api/tags.json")) as { tags: MinimalTag[] }
```

Comment:
You can just return the promise here, no need to await it.

```suggestion
        const fetchTags = async () => 
            admin.getJSON<{ tags: MinimalTag[] }>("/api/tags.json")
```